### ISSUE TITLE
Comportement des entités

### ISSUE THREAD

**Issue by AnteJR** [AnteJR/UNIL-GameJam#2]
Les entités "ennemies" doivent avoir 3 patterns de comportement dans leur déplacement horizontal : 

1. Aller lentement de l'autre côté de la route
2. Aller vite de l'autre côté de la route
3. Avoir un rythme variable/saccadé pour traverser la route

Pour le moment, c'est Florian qui se voit assigner cette tâche, avec Isaac en renfort si besoin (notamment pour l'usage de fonction Kaboom qui allègeraient le code).

**Comment by astonedf** [comment]
Je pense qu'il faudrait aussi varier la position x des entités pour qu'elles apparaissent déjà au milieu de la route par exemple (mais pas qu'elles aient déjà traversé la route)

**Comment by astonedf** [comment]
Il faut que le déplacement des entités soit adapté à la proportion aussi, pour l'instant ça change sur mobile et ordi (d'ailleurs sur mobile les entités traversent tous avant que le mouton arrive donc il n'y a quasi pas de collision.

**Comment by florian-rieder** [comment]
Effectivement, le timing des obstacles fait que selon comment on joue (toujours appuyé ou toujours non appuyé), fait qu'on évite assez facilement tous les obstacles, sans devoir faire grand chose. Peut-être qu'on pourrait envisager un système un peu plus actif, où les obstacles choisissent leur délai de départ pour essayer d'intercepter le joueur en fonction de sa vitesse au moment où l'obstacle est spawné, avec un certain degré d'aléatoire ?

**Comment by AnteJR** [comment]
Ce serait next level, et probablement le plus idéal pour du challenge, mais à voir si ça nous rajoute pas trop de complexité.

**Comment by florian-rieder** [comment]
On peut sans doute au moins prendre en compte la hauteur de l'écran pour régler les délais de départ, histoire que ce soit pareil sur tous les écrans. Je vais voir ce que je peux faire !

**Comment by astonedf** [comment]
Je pense pas que la solution soit dans l'aléatoire. Par contre un truc à garder en tête c'est qu'il faut que ça soit possible d'obtenir toutes les cartes si on joue bien, pour l'instant ça manque encore ce côté skillé. J'ai l'impression que peu importe ce que je fais, je les chope quasi tous si je laisse appuyé ou les rates quasi tous en essayant de freiner et d'accelérer donc c'est justement trop aléatoire encore

**Comment by florian-rieder** [comment]
Pour le côté skill, au delà des comportements des obstacles, on peut aussi faire en sorte que quand les amis apparaissent, ce soit au bord de la route, qu'ils attendent un petit moment, puis se déplacent vers le centre de la route. Les timings sont à déterminer pour qu'on puisse toujours l'atteindre d'une manière ou d'une autre, mais ça ferait qu'il faudrait parfois temporiser un peu pour attraper la carte.

**Comment by ipante** [comment]
> Pour le côté skill, au delà des comportements des obstacles, on peut aussi faire en sorte que quand les amis apparaissent, ce soit au bord de la route, qu'ils attendent un petit moment, puis se déplacent vers le centre de la route. Les timings sont à déterminer pour qu'on puisse toujours l'atteindre d'une manière ou d'une autre, mais ça ferait qu'il faudrait parfois temporiser un peu pour attraper la carte.

C'est en effet ainsi que j'imaginais les porteurs de courrier : comme venant du bord de la carte et nous forçant à trouver un le bon timing pour les rejoindre. Le but du jeu étant de heurter les bons et d'éviter les mauvais. Seulement, pour que ça marche, il faut que le mouton livreur passe. Or il y a un problème de narration à résoudre : pourquoi le mouton livreur ne nous attendrait pas, une fois arrivé au bord de la route ? Sarah l'avait déjà évoqué lors de la Jam.

**Comment by ipante** [comment]
Une solution purement narrative :

1. passé un certain temps, un chien débarque et fait détaler le mouton (qui quitte donc le bord de route et ne remettra pas le courrier).
2. le mouton est toujours poursuivi par un chien : il faut donc vraiment le croiser au bon moment.

**Comment by astonedf** [comment]
C'est vraiment super les moutons qui se déplacent avec la lettre!!
Je pense que ça serait bien d'en avoir quelques uns qui bouge pas, surtout au tout début (impossible à rater) pour montrer qu'on doit collecter les lettres et peut-être 2 autres plus tard pour que ceux qui galèrent plus puissent finir avec plus qu'une lettre haha
